Update auto-update cfg & add homepage, missing files of blueimp-gallery

sufuf3 opened the pull request:
close #11717
Pull request for issue: #11717
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/blueimp/Gallery
 * Official website (optional, not the repository): https://blueimp.github.io/Gallery/
 * NPM package url (optional): https://www.npmjs.com/package/blueimp-gallery
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 218
   - Count of stars: 2509
   - Count of forks: 824
 * NPM download stats (optional):
   - Downloads in the last day: 105
   - Downloads in the last week: 839
   - Downloads in the last month: 3,319

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [ ] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 800 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

extend1994 commented on ajax/libs/blueimp-gallery/package.json:
```diff
@@ -36,27 +36,8 @@
-        "js/jquery.blueimp-gallery.js",
-        "js/jquery.blueimp-gallery.min.js"
+        "img/*",
+        "+(css|js)/*blueimp*"
```
Let's use `+(css|js)/*blueimp*.+(css|map|js)` to avoid .less file ,which appeared before v2.21.3, appears again in the future.

sufuf3 commented:
@extend1994 I updated the commit message, please help me review this PR, thank you.

sufuf3 commented:
@extend1994 I updated the commit message, please help me review this PR, thank you.

sufuf3 commented:
@maruilian11 @pvnr0082t 
Please help me review this PR, Thank you.

maruilian11 commented:
@sufuf3 Sorry for the late reply, I think you may need to add files for v0.26.0 and v0.27.0, too.

sufuf3 commented:
@maruilian11 I added the missing files for v2.26.0 and v2.27.0 
these two versions release 15 days ago 😞 
please help me review this PR. Thanks

PS. After this comment, if there is any version release, and the auto-update is fine, just need to be added the missing files then I'll add a new commit or PR to do this.
Because it's good to update the auto-update config ASAP and we don't need to pending on this PR and always manually add the missing files looply. (waiting for reviewing again and again because this PR need to be added the missing files of newest versions) 
Also, we can use newest auto-update config to get the release versions to prevent manually add the missing files looply. And manually add the missing files of old versions to let this lib better.

extend1994 commented:
@sufuf3 I think we don't need to manually add v0.26.0 and v0.27.0. They meet auto-update config.

sufuf3 commented:
@extend1994 
https://github.com/cdnjs/cdnjs/tree/master/ajax/libs/blueimp-gallery/2.26.0
https://github.com/cdnjs/cdnjs/tree/master/ajax/libs/blueimp-gallery/2.27.0
here is the v2.26.0 and v2.27.0, I don't know what you mean?
It really needs to be added the missing files of v2.26.0 and v2.27.0, isn't it?

sufuf3 commented:
@maruilian11 @cdnjs/active-contributor @cdnjs/intern3 
Please help me review this PR, Thank you.

PS. I hope this PR can be merged before the lib release the new version.
Thanks.

extend1994 commented on ajax/libs/blueimp-gallery/package.json:
```diff
@@ -1,7 +1,7 @@
 {
   "name": "blueimp-gallery",
   "filename": "js/blueimp-gallery.min.js",
-  "version": "2.27.0",
```
Should this be touched?

sufuf3 commented:
@extend1994 I revised the version field
Please help me review this PR, Thank you.

extend1994 commented:
You have not added `img/play.png` in v2.0.0 ~ v2.1.0 I mentioned?

sufuf3 commented:
@extend1994 @maruilian11 I added img/play.png in v2.0.0 ~ v2.1.0
Please help me review this PR, Thank you.

sufuf3 commented:
@maruilian11 I added `2.12.4, 2.11.5, 2.3.4, 2.3.0, 2.2.0` versions from GitHub.
Please help me review this PR, thanks.

maruilian11 commented on ajax/libs/blueimp-gallery/package.json:
```diff
@@ -36,27 +36,8 @@
-        "js/jquery.blueimp-gallery.js",
-        "js/jquery.blueimp-gallery.min.js"
+        "img/*",
+        "+(css|js)/*blueimp*.+(css|map|js)"
```
`"+(css|js)/**/blueimp*.+(css|map|js)"` would be better.

sufuf3 replied:
I think that excludes `dist` and `build` as the basePath... we don't have to use `**/*`.
Such as https://github.com/cdnjs/cdnjs/blob/master/ajax/libs/imagehover.css/package.json#L25
https://github.com/cdnjs/cdnjs/blob/master/ajax/libs/css-social-buttons/package.json#L26
Am I wrong?

Thank you.

sufuf3 replied:
@maruilian11 I think that excludes `dist` and `build` as the basePath... we don't have to use `**/*`.
Such as https://github.com/cdnjs/cdnjs/blob/master/ajax/libs/imagehover.css/package.json#L25
https://github.com/cdnjs/cdnjs/blob/master/ajax/libs/css-social-buttons/package.json#L26
Am I wrong?

Thank you.

maruilian11 replied:
OK. I think both are fine.

maruilian11 commented:
Mention the minify part?

sufuf3 commented:
@maruilian11 I mentioned it.
@cdnjs/active-contributor @cdnjs/intern3 @cdnjs/library-reviewer @extend1994 
Please help me review this PR, thanks.

maruilian11 commented:
Also add the `homepage`? https://blueimp.github.io/Gallery/

sufuf3 commented:
@maruilian11 I added homepage info
Please help me review this PR, thanks.

sufuf3 commented:
@cdnjs/library-reviewer Please help me review this PR, thank you.

sufuf3 commented:
@maruilian11 @extend1994 Please help me review this PR, thank you.

sufuf3 commented:
ping @maruilian11 @extend1994

sufuf3 commented:
ping @maruilian11 @extend1994

sufuf3 commented:
@PeterDaveHello sorry that this PR waiting for reviewer's review for a while, almost a month.
But this is a high priority PR.
So I request you to help review this PR.

Please help me review this PR, thanks.

sufuf3 commented:
ping @maruilian11 @extend1994

sufuf3 commented:
Added missing files as the following in first commit & rebased with the latest master branch.
- ajax/libs/blueimp-gallery/2.28.0/css/blueimp-gallery.min.css.map
- ajax/libs/blueimp-gallery/2.28.0/js/blueimp-gallery.min.js.map
- ajax/libs/blueimp-gallery/2.28.0/js/jquery.blueimp-gallery.min.js.map
- ajax/libs/blueimp-gallery/2.29.0/css/blueimp-gallery.min.css.map
- ajax/libs/blueimp-gallery/2.29.0/js/blueimp-gallery.min.js.map
- ajax/libs/blueimp-gallery/2.29.0/js/jquery.blueimp-gallery.min.js.map

sufuf3 commented:
ping @extend1994

sufuf3 commented:
I think we can add `ajax/libs/blueimp-gallery/2.30.0/*` missing files after this PR be merged.
Because I do this action many time.
I think merge this PR and let the auto-update config work are better than manually add the missing files again and again.
Thanks.

sufuf3 commented:
@extend1994 I updated the commit message, also added the files as the following:
```
 create mode 100644 ajax/libs/blueimp-gallery/2.30.0/css/blueimp-gallery.min.css.map
 create mode 100644 ajax/libs/blueimp-gallery/2.30.0/js/blueimp-gallery.min.js.map
 create mode 100644 ajax/libs/blueimp-gallery/2.30.0/js/jquery.blueimp-gallery.min.js.map
```

Hope I won't add the missing files in this PR again in future because I added them again and again. 😞  
In fact, we can stop this via update the git auto-update config. 😃 
Thanks.

extend1994 commented:
@PeterDaveHello Please help review this PR, thank you.

sufuf3 commented:
@PeterDaveHello Would you please let me know how could I improve this PR?
Many thanks.

sufuf3 commented:
I would close this PR.
Because I found this lib update to v2.31.0 recently, but this PR is waiting for reviewing for a while since 1 month ago.
I need to check the lib in my free time these days.
Thanks

sufuf3 commented:
I updated the first commit message from `v2.30.0` to `v2.32.0`, also added the files as the following:

```
        new file:   ajax/libs/blueimp-gallery/2.31.0/css/blueimp-gallery.min.css.map
        new file:   ajax/libs/blueimp-gallery/2.31.0/js/blueimp-gallery.min.js.map
        new file:   ajax/libs/blueimp-gallery/2.31.0/js/jquery.blueimp-gallery.min.js.map
        new file:   ajax/libs/blueimp-gallery/2.31.1/css/blueimp-gallery.min.css.map
        new file:   ajax/libs/blueimp-gallery/2.31.1/js/blueimp-gallery.min.js.map
        new file:   ajax/libs/blueimp-gallery/2.31.1/js/jquery.blueimp-gallery.min.js.map
        new file:   ajax/libs/blueimp-gallery/2.32.0/css/blueimp-gallery.min.css.map
        new file:   ajax/libs/blueimp-gallery/2.32.0/js/blueimp-gallery.min.js.map
        new file:   ajax/libs/blueimp-gallery/2.32.0/js/jquery.blueimp-gallery.min.js.map
```

Because bot added  v2.31.0 ~ v2.32.0 during this PR wait for the review.
@PeterDaveHello Would you please review this PR in your free time?
Because this PR is high priority. Also, if this PR can be merged, we don't have to add these missing files again and again.
Many thanks.

sufuf3 commented:
I updated the first commit message from `v2.30.0` to `v2.32.0`, also added the files as the following:

```
        new file:   ajax/libs/blueimp-gallery/2.31.0/css/blueimp-gallery.min.css.map
        new file:   ajax/libs/blueimp-gallery/2.31.0/js/blueimp-gallery.min.js.map
        new file:   ajax/libs/blueimp-gallery/2.31.0/js/jquery.blueimp-gallery.min.js.map
        new file:   ajax/libs/blueimp-gallery/2.31.1/css/blueimp-gallery.min.css.map
        new file:   ajax/libs/blueimp-gallery/2.31.1/js/blueimp-gallery.min.js.map
        new file:   ajax/libs/blueimp-gallery/2.31.1/js/jquery.blueimp-gallery.min.js.map
        new file:   ajax/libs/blueimp-gallery/2.32.0/css/blueimp-gallery.min.css.map
        new file:   ajax/libs/blueimp-gallery/2.32.0/js/blueimp-gallery.min.js.map
        new file:   ajax/libs/blueimp-gallery/2.32.0/js/jquery.blueimp-gallery.min.js.map
```

Because bot added  v2.31.0 ~ v2.32.0 during this PR wait for the review.
@PeterDaveHello Would you please review this PR in your free time?
Because this PR is high priority. Also, if this PR can be merged, we don't have to add these missing files again and again.
Many thanks.

PeterDaveHello commented:
@sufuf3 yeah, sure, sorry for the delay.
